### DISCUSSION TITLE
Modifying the number of input channels for the encoder

### DISCUSSION THREAD

**Discussion by kiraicode** [open-mmlab/mmclassification#1377] [Q&A]
Hi all, thanks in advance for your help! I have a task where my input images are in the shape 5xWxH. I would like to use some of the backbones in this library but I'm not entirely sure how I can modify them to take my custom images as input. Or is this supported by default? If so, I may have missed it in the documentation.

Ideally, I would like to use a transformer-based backbone that can support 5xHxW inputs.

Any pointers would be of great help!

**Comment by Ezra-Yu** [comment]
most of the backbones have `in_channels` like [resnet](https://github.com/open-mmlab/mmclassification/blob/fb16bdc6a25f96843f8fb97b02e45872c61dc566/mmcls/models/backbones/resnet.py#L410), vit, swin and etc.

```
# model settings
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='ResNet',
        depth=18,
        num_stages=4,
        in_channels=3,
        out_indices=(3, ),
        style='pytorch'),
    neck=dict(type='GlobalAveragePooling'),
    head=dict(
        type='LinearClsHead',
        num_classes=1000,
        in_channels=512,
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, 5),
    ))
```

There are still some infrequently-used backbones without these params, you need to do some modifications. If you do that, PRs are welcome.
